### ISSUE TITLE
Loss in compile question.

### ISSUE THREAD

**Issue by asimay** [matterport/Mask_RCNN#1026]
why https://github.com/matterport/Mask_RCNN/blob/master/mrcnn/model.py#L2188

loss is defined to multiply by length of output?? it's weired.. Could you please explain it? thank you very much!

**Comment by keineahnung2345** [comment]
There will be 5 losses for Mask RCNN. `loss=[None] * len(self.keras_model.outputs)` holds the place used later.

**Comment by shikunyu8** [comment]
@keineahnung2345  When will the hold places be used? The five losses and regularization loss are added before this line.


```       
         loss_names = [
            "rpn_class_loss",  "rpn_bbox_loss",
            "mrcnn_class_loss", "mrcnn_bbox_loss", "mrcnn_mask_loss"]
        for name in loss_names:
            layer = self.keras_model.get_layer(name)
            if layer.output in self.keras_model.losses:
                continue
            loss = (
                tf.reduce_mean(layer.output, keepdims=True)
                * self.config.LOSS_WEIGHTS.get(name, 1.))
            self.keras_model.add_loss(loss)

        # Add L2 Regularization
        # Skip gamma and beta weights of batch normalization layers.
        reg_losses = [
            keras.regularizers.l2(self.config.WEIGHT_DECAY)(w) / tf.cast(tf.size(w), tf.float32)
            for w in self.keras_model.trainable_weights
            if 'gamma' not in w.name and 'beta' not in w.name]
        self.keras_model.add_loss(tf.add_n(reg_losses))

        # Compile
        self.keras_model.compile(
            optimizer=optimizer,
            loss=[None] * len(self.keras_model.outputs))
```

**Comment by SimchaGD** [comment]
Hi I am wondering the same thing. Is there already an answer?
